### ISSUE TITLE
PhantomJS disconnects/crashes after completing all tests

### ISSUE THREAD

**Issue by tonyfinn** [karma-runner/karma#2012]
### Expected behavior
1. Karma runs all tests and reports overall success/failure at the end.

### Actual behavior
1. Karma runs all tests (can see the console reporter reporting progress)
2. Waits browerNoActivityTimeout
3. Reports PhantomJS as disconnected.
4. Reports the tests overall as failed

### Enviroment Details

- Karma version (output of `karma --version`): 0.13.22
- Relevant part of your `karma.config.js` file

    browsers: ['PhantomJS'],
    captureTimeout: 60000,
    singleRun: true,

### Steps to reproduce the behaviour

I've had a lot of difficulty narrowing it down. It started when adding a new test case for some very simple code, but if I run just that test case without or other 900 tests it passes. If I run every test except this one it passes.

The test case in question causes the build to fail if included with other tests even if I comment out the test body:

    it('for scheduled flight with change in pacing should return [budget, pacing, startDate, endDate]', function () {
       /*flightService.save(saveData, saveData, {pacing: saveData.pacing});
       $rootScope.$digest();
       var validateProperties =  ['budget', 'pacing', 'startDate', 'endDate'];
       expect(commonValidationService.validate).toHaveBeenCalled();
       expect(commonValidationService.validate.calls.mostRecent().args[1]).toEqual(validateProperties);
       expect(restService.put).toHaveBeenCalled();*/
    });

+ entire rest of the test suite crashes PhantomJS at the end of the test suite.

while:

    xit('for scheduled flight with change in pacing should return [budget, pacing, startDate, endDate]', function () {
        /*flightService.save(saveData, saveData, {pacing: saveData.pacing});
       $rootScope.$digest();
       var validateProperties =  ['budget', 'pacing', 'startDate', 'endDate'];
       expect(commonValidationService.validate).toHaveBeenCalled();
       expect(commonValidationService.validate.calls.mostRecent().args[1]).toEqual(validateProperties);
       expect(restService.put).toHaveBeenCalled();*/
    });

does not.

Any suggestions as to where I could start debugging this further would be very appreciated.

**Comment by chrismagri** [comment]
i have the same issue as yours..

**Comment by dignifiedquire** [comment]
Does the suite pass in another browser? If so it's probably some issue related to PhantomJS and its memory/resource constraints.

**Comment by laurelnaiad** [comment]
I see PhantonJS crash if I set `phantomjsLauncher.exitOnResourceError` to true in my config, even when there are no reported resource errors.

**Comment by tonyfinn** [comment]
So we looked into this a bit more in our project. Checking in other browsers found that one of our dependencies (Pendo specifically) was making a HTTP request in it's config method. The test suite in question loaded a angular module that indirectly depended on this module in its beforeEach(). After 41 tests that called this module (=41 http requests), Phantom seemed to just fall over.

We just stubbed it out in our unit tests since that wasn't specifically what we were trying to test. Now there's no HTTP requests being made and we are no longer getting the disconnected issue.
